x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/utils/get_engines_display_text.tsx

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { ADMIN, ALL } from '../constants';
+import { IApiToken } from '../types';
+
+export const getEnginesDisplayText = (apiToken: IApiToken): JSX.Element | string => {
+  const { type, access_all_engines: accessAll, engines = [] } = apiToken;
+  const engineList = () => (
+    <ul className="credentials-engine-list">
+      {engines.map((engine) => (
+        <li key={engine} className="credentials-engine-list__engine">
+          {engine}
+        </li>
+      ))}
+    </ul>
+  );
+
+  if (type === ADMIN) {
+    return '--';
+  }
+  return accessAll ? ALL : engineList();
```

Comment:
I think we can rewrite this return flow to be a little simpler to read - let me know what you think!

```suggestion
  const { type, access_all_engines: accessAll, engines = [] } = apiToken;

  if (type === ADMIN) {
    return '--';
  }
  if (accessAll) {
    return ALL;
  }
  return (
    <ul>
      {engines.map((engine) => (
        <li key={engine}>{engine}</li>
      ))}
    </ul>
  );
```

What do you think? Does that branching path feel more straightforward to you?

Reply:
This, like the majority of code you are reviewing, is largely copy / paste from the old codebase. I focused more on isolating units and adding unit tests. Happy to make this change though.

Reply:
Totally get that btw! I want to make sure all my suggestions are actual improvements on the old code, so I'm always looking for confidence checks (+ happy to take pushback if people prefer them as-is) 👍

Reply:
That's the beauty of unit tests right? It's easy enough to refactor this stuff and be confident it still behaves like we need. This reads cleaner to me and the specs pass so 💯

Reply:
🎉 🎉 🎉

Reply:
564737a